libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -23,18 +23,23 @@ logScope:
   topic = "PubSubPeer"
 
 type
-    PubSubPeer* = ref object of RootObj
-      proto: string # the protocol that this peer joined from
-      sendConn: Connection
-      peerInfo*: PeerInfo
-      handler*: RPCHandler
-      topics*: seq[string]
-      sentRpcCache: TimedCache[string] # cache for already sent messages
-      recvdRpcCache: TimedCache[string] # cache for already received messages
-      refs*: int # refcount of the connections this peer is handling
-      onConnect: AsyncEvent
-
-    RPCHandler* = proc(peer: PubSubPeer, msg: seq[RPCMsg]): Future[void] {.gcsafe.}
+  PubSubObserver* = ref object
+    onRecv*: proc(peer: PubSubPeer; msgs: var RPCMsg) {.gcsafe.}
```

Comment:
Usage of `PubSubPeer` do not help observer to get any useful information, because `PubSubPeer` stores only internal data and `PeerInfo`, so it can be easily replaced with `PeerInfo`.

Reply:
From the outside you can ignore PubSubPeer, it is just an implementation detail, just check the onSend/onRecv